File: z-clients/javascript/src/models/adminAuthTokenExpireIn.ts
// this file is @generated
import { Temporal } from 'temporal-polyfill-lite';

export interface AdminAuthTokenExpireIn {
    id: string;
    /** Milliseconds from now until the token expires. `None` means expire immediately. */
    expiry?: Temporal.Duration | null;
}

export const AdminAuthTokenExpireInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AdminAuthTokenExpireIn {
        return {
            id: object['id'],
            expiry: object['expiry_ms'] != null ? Temporal.Duration.from({ milliseconds: object['expiry_ms'] }) : undefined,
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AdminAuthTokenExpireIn): any {
        return {
            'id': self.id,
            'expiry_ms': self.expiry != null ? self.expiry.total('millisecond') : undefined,
        };
    }
}

File: z-clients/python/diom/models/admin_auth_token_expire_in.py
# this file is @generated
from pydantic import Field

from ..internal.base_model import BaseModel
from ..internal.types import TimeDeltaMs


class AdminAuthTokenExpireIn(BaseModel):
    id: str

    expiry: TimeDeltaMs | None = Field(alias="expiry_ms", default=None)
    """Milliseconds from now until the token expires. `None` means expire immediately."""
